### ISSUE TITLE
Gradient accessor with FSDP2

### ISSUE THREAD

**Issue by K-H-Ismail** [pytorch/torchtitan#1162]
Dear developers,

For debugging purposes, I want to access the gradient of one of the weights in a model during training using the train config for llama3.1 8b. For example, let's say this weight:
https://github.com/pytorch/torchtitan/blob/e4ae55b381e560913b3e1d8b9403f226e7bf2629/torchtitan/models/llama3/model.py#L297

Do you know how I can do this?
Since fully_shard() has removed `use_orig_params`, printing the `.grad` returns None. Implementing a hook also returns None. I've tried doing a `.full_tensor().grad` on the weight DTensor, but it still doesn't return the grad. Could you please provide a hint here?

**Comment by fegin** [comment]
`use_orig_params` is removed because that's the only implementation for `fully_shard()`. You can access the the gradient through `.grad`.  You can also check `torchtitan.distributed.utils.clip_grad_norm_`.

**Comment by K-H-Ismail** [comment]
It works, thanks for your help!
